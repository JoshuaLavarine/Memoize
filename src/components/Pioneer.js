import React, { Component } from 'react';
import Article from './Article';

class Pioneer extends Component {
  constructor() {
    super();
    this.state = {
      currentPioneer: 0
    }
  }

  selectPioneer = (event) => {
    this.setState({
      currentPioneer: event.target.id
    })
  }

  displayPioneers = () => {
    return this.props.pioneers.map(pioneer => {
      return (
        <section key={pioneer.pioneer} className="pioneer-card">
          <img alt="" className="pioneer-img" onClick={this.selectPioneer} src={pioneer.img} id={pioneer.id}></img>
          <p>{pioneer.pioneer}</p>
        </section>
      )
    })
  } 

  render() {
    switch(this.state.currentPioneer > 0) {
      case(true):
      return (
        <Article 
        currentPioneer={this.state.currentPioneer}
        pioneers={this.props.pioneers}
        selectPioneer = {this.selectPioneer}
        />  
      )
    default:
      return (
        <article className="pioneer-layout">
          <p>Please select a pioneer's image to learn more.</p>
          <section className="contain-pioneers">
            <section className="individual-pioneers"> 
              {this.displayPioneers()}
            </section>
          </section>
        </article>
      )
    }
  }
}

export default Pioneer