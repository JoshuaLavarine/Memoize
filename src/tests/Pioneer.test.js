import React from "react";
import Pioneer from "../Components/Pioneer";
import { shallow } from "enzyme";
import { pioneers } from "./mockData";

describe("Pioneer", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Pioneer 
      pioneers={pioneers}
      />   
    );
  });
	
  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a proper default state", () => {
    expect(wrapper.state()).toEqual({ 
			currentPioneer: 0
		});
	});
});