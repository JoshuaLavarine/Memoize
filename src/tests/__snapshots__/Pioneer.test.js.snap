// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pioneer should match the snapshot with all data passed in 1`] = `
<article
  className="pioneer-layout"
>
  <p>
    Please select a pioneer's image to learn more.
  </p>
  <section
    className="contain-pioneers"
  >
    <section
      className="individual-pioneers"
    >
      <section
        className="pioneer-card"
        key="Grace Hopper"
      >
        <img
          alt=""
          className="pioneer-img"
          id={1}
          onClick={[Function]}
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/ad/Commodore_Grace_M._Hopper%2C_USN_%28covered%29.jpg/440px-Commodore_Grace_M._Hopper%2C_USN_%28covered%29.jpg"
        />
        <p>
          Grace Hopper
        </p>
      </section>
      <section
        className="pioneer-card"
        key="Augusta Ada King, Countess of Lovelace"
      >
        <img
          alt=""
          className="pioneer-img"
          id={2}
          onClick={[Function]}
          src="https://upload.wikimedia.org/wikipedia/commons/0/0b/Ada_Byron_daguerreotype_by_Antoine_Claudet_1843_or_1850.jpg"
        />
        <p>
          Augusta Ada King, Countess of Lovelace
        </p>
      </section>
      <section
        className="pioneer-card"
        key="Alan Turing"
      >
        <img
          alt=""
          className="pioneer-img"
          id={3}
          onClick={[Function]}
          src="https://i.redd.it/51dcpxhdwcox.jpg"
        />
        <p>
          Alan Turing
        </p>
      </section>
      <section
        className="pioneer-card"
        key="Timothy John Berners-Lee"
      >
        <img
          alt=""
          className="pioneer-img"
          id={4}
          onClick={[Function]}
          src="https://upload.wikimedia.org/wikipedia/commons/4/4e/Sir_Tim_Berners-Lee_%28cropped%29.jpg"
        />
        <p>
          Timothy John Berners-Lee
        </p>
      </section>
      <section
        className="pioneer-card"
        key="Jennifer Pahlka"
      >
        <img
          alt=""
          className="pioneer-img"
          id={5}
          onClick={[Function]}
          src="http://www.greatthoughtstreasury.com/sites/default/files/Jennifer-Pahlka1%5B1%5D.jpg"
        />
        <p>
          Jennifer Pahlka
        </p>
      </section>
    </section>
  </section>
</article>
`;
